packages/react/src/components/TreeView/TreeView.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { keys, match, matches } from '../../internal/keyboard';
+
+const { prefix } = settings;
+
+export default function TreeView({
+  children,
+  className,
+  multiselect,
+  onChange,
+  selected: preselected = [],
+  size = 'default',
+  ...rest
+}) {
+  const treeClasses = classNames(className, `${prefix}--tree`, {
+    [`${prefix}--tree--${size}`]: size !== 'default',
+  });
+  const treeRootRef = useRef(null);
+  const treeWalker = useRef(treeRootRef?.current);
+  const [selected, setSelected] = useState(preselected);
+  const handleSelect = (event, { value } = {}) => {
+    if (multiselect && event.metaKey) {
+      if (!selected.includes(value)) {
+        setSelected(selected.concat(value));
+      } else {
+        setSelected(selected.filter(selectedValue => selectedValue !== value));
+      }
+    } else {
+      setSelected([value]);
+    }
+    if (onChange) {
+      onChange(event, { value: value?.value });
+    }
+  };
+  const nodesWithProps = React.Children.map(children, node => {
+    if (React.isValidElement(node)) {
+      return React.cloneElement(node, {
+        depth: 0,
+        onSelect: handleSelect,
+        selected,
+      });
+    }
+  });
+
+  const handleKeyDown = event => {
+    event.stopPropagation();
+    if (matches(event, [keys.ArrowUp, keys.ArrowDown])) {
+      event.preventDefault();
+    }
+    treeWalker.current.currentNode = document.activeElement;
+    if (match(event, keys.ArrowUp)) {
+      treeWalker.current.previousNode()?.focus();
+    }
+    if (match(event, keys.ArrowDown)) {
+      treeWalker.current.nextNode()?.focus();
+    }
```

Comment:
Seems that there is no visual cue to tell somewhere in treeview got focus. I believe it was not intentional?

```suggestion
    let nextFocusNode;
    if (match(event, keys.ArrowUp)) {
      nextFocusNode = treeWalker.current.previousNode();
    }
    if (match(event, keys.ArrowDown)) {
      nextFocusNode = treeWalker.current.nextNode();
    }
    if (nextFocusNode && nextFocusNode !== event.target) {
      Array.prototype.forEach.call(treeRootRef?.current?.querySelectorAll('[tabIndex="0"]') ?? [], item => {
        item.tabIndex = -1;
      });
      nextFocusNode.tabIndex = 0;
      nextFocusNode.focus();
    }
```